src/ripple/RippleEngine.ts

```diff
@@ -119,6 +138,92 @@ export class XrpEngine extends CurrencyEngine<
     this.networkInfo = networkInfo
     this.nonce = 0
     this.minimumAddressBalance = this.networkInfo.baseReserve
+    this.otherMethods = {
+      makeTx: async (params: MakeTxParams) => {
+        if (params.type === 'MakeTxDexSwap') {
+          const {
+            expiration,
+            metadata,
+            fromNativeAmount,
+            fromTokenId,
+            toNativeAmount,
+            toTokenId
+          } = params
+
+          const TakerGets = this.nativeToXrpAmount(
+            fromNativeAmount,
+            fromTokenId
+          )
+          const TakerPays = this.nativeToXrpAmount(toNativeAmount, toTokenId)
+
+          const Expiration = unixTimeToRippleTime(
+            (expiration ?? DEX_ORDER_EXPIRATION_TIME_DEFAULT_S) * 1000
+          )
```

Comment:
Optional: I see that you are using uppercase, because that is what the `OfferCreate` JSON expects. However, we generally reserve uppercase names for types. Would it make more sense to use lowercase for the variables, and then do `{ ... Expiration: exipration, ... }` and such to change case inside the object?

Reply:
it crossed my mind but went for the convenient route. will do